File: Python/RemoveLinkedListElements.py
from typing import Optional

class ListNode:
    def __init__(self,val):
        self.val=val
        self.next=None
        
class Solution:
    def removeElements(self, head:Optional [ListNode],val : int) -> Optional[ListNode]:
        dummy=ListNode(0)
        dummy.next=head
        curr=dummy
        while curr.next:
            if curr.next.val==val:
                curr.next=curr.next.next
            else:
                curr=curr.next
        return dummy.next
    
def print_ll(head):
    temp=head
    while temp:
        print(temp.val,end=' -> ')
        temp=temp.next
    print(None)
def listToLL(ls):
    head=ListNode(ls[0])
    curr=head
    for i in (ls[1:]):
        curr.next=ListNode(i)
        curr=curr.next
    return head

s=Solution()
head=listToLL([1,2,6,3,4,5,6])
val=6
print_ll(head)
head=s.removeElements(head,val)
print_ll(head)

File: Java/RemoveLinkedListElements.java
class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }
}

class RemoveLinkedListElements{
    public ListNode removeElements(ListNode head,int val){
        ListNode dummy=new ListNode(0);
        dummy.next=head;
        ListNode curr=dummy;
        while(curr.next != null){
            if(curr.next.val== val){
                curr.next=curr.next.next;
            }else{
                curr=curr.next;
            }
        }
        return dummy.next;
    }

    public ListNode printLL(ListNode head){
        ListNode curr=head;
        while (curr!=null){
            System.out.print(curr.val + " -> ");
            curr=curr.next;
        }
        System.out.println("null");
        return head;
    }
    public static void  main(String[] args){
        ListNode head = new ListNode(1, new ListNode(2, new ListNode(6,
                        new ListNode(3, new ListNode(4, new ListNode(5, new ListNode(6)))))));

        RemoveLinkedListElements obj = new RemoveLinkedListElements();
        System.out.print("Original List: ");
        obj.printLL(head);

        head = obj.removeElements(head, 6);
        System.out.print("After Removing 6: ");
        obj.printLL(head);
    }
}